### PR TITLE
Add tests for ClientExceptionFactory

### PR DESCRIPTION
**Pull request by sancar**

### CODE REVIEW

**Comment by ihsandemir** [comment]
:+1:

File: hazelcast/src/test/java/com/hazelcast/client/protocol/ClientExceptionFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+    }
+
+    @Test
+    public void testException() {
+        ClientMessage exceptionMessage = exceptionFactory.createExceptionMessage(throwable);
+        ClientMessage responseMessage = ClientMessage.createForDecode(exceptionMessage.buffer(), 0);
+        Throwable resurrectedThrowable = exceptionFactory.createException(responseMessage);
+        assertEquals(throwable.getClass(), resurrectedThrowable.getClass());
+        assertArray(throwable.getStackTrace(), resurrectedThrowable.getStackTrace());
```
Comment (cangencer) [inline hazelcast/src/test/java/com/hazelcast/client/protocol/ClientExceptionFactoryTest.java]: we can use assertArrayEquals here instead

Comment (sancar) [inline hazelcast/src/test/java/com/hazelcast/client/protocol/ClientExceptionFactoryTest.java]: Thanks. I did not know the exact naming, and my IDE did not suggest when I am trying to find out. Changing it now.

**Comment by cangencer** [comment]
:+1:

**Pull request merged**